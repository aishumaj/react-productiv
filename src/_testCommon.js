const testTodo = {id:1, title:"hello", description: "test", priority:1};

const testTodos = [{id:1, title:"hello", description: "test", priority:2},
  {id:1, title:"top priority", description: "test2", priority:1},
  {id:1, title:"hello", description: "test3", priority:3}]

export {testTodo, testTodos};