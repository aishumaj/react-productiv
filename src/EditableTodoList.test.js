import React from "react";
import { fireEvent, render } from "@testing-library/react";
import EditableTodoList from "./EditableTodoList";
import { testTodos } from "./_testCommon";

let mock = jest.fn();

describe("EditableTodoList app", function () {
  it("renders without crashing", function () {
    render(<EditableTodoList todos={testTodos} update={mock} remove={mock} />);
  });

  it("displays editable todo list", function () {
    const result =
      render(<EditableTodoList todos={testTodos} update={mock} remove={mock} />);

    expect(result.queryByText("test")).toBeInTheDocument();
    expect(result.queryByText("test2")).toBeInTheDocument();
    expect(result.queryByText("test3")).toBeInTheDocument();
    expect(result.container.querySelectorAll(".EditableTodo-toggle").length).toEqual(3);
    expect(result.container.querySelectorAll(".EditableTodo-delBtn").length).toEqual(3);
  });

  it("displays editable form when edit clicked", function () {
    const { container } =
      render(<EditableTodoList todos={testTodos} update={mock} remove={mock} />);
    const editBtn = container.querySelector(".EditableTodo-toggle");

    fireEvent.click(editBtn);
    expect(editBtn).not.toBeInTheDocument();
    expect(container.querySelectorAll(".EditableTodo-toggle").length).toEqual(2);
    expect(container.querySelector(".NewTodoForm")).toBeInTheDocument();
    expect(container.querySelector("#newTodo-title")).toContainHTML("hello");
  });

  test("matches snapshot", function () {
    const { container } =
      render(<EditableTodoList todos={testTodos} update={mock} remove={mock} />);
    expect(container).toMatchSnapshot();
  });

});