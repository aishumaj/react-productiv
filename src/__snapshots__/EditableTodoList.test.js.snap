// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableTodoList app matches snapshot 1`] = `
<div>
  <div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
          id="1"
        >
          <div>
            <b>
               
              hello
               
            </b>
             
            <small>
               (priority: 
              2
              )
            </small>
          </div>
          <div>
            <small>
              test
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
          id="1"
        >
          <div>
            <b>
               
              top priority
               
            </b>
             
            <small>
               (priority: 
              1
              )
            </small>
          </div>
          <div>
            <small>
              test2
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
          id="1"
        >
          <div>
            <b>
               
              hello
               
            </b>
             
            <small>
               (priority: 
              3
              )
            </small>
          </div>
          <div>
            <small>
              test3
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
